pex/pex_info.py

```diff
@@ -451,11 +451,18 @@ def raw_pex_root(self):
     @property
     def pex_root(self):
         # type: () -> str
-        pex_root = os.path.expanduser(self.raw_pex_root)
-        if not can_write_dir(pex_root):
+        paths = os.path.expanduser(self.raw_pex_root).split(",")
+        pex_root = None
+        writable = False
+        for path in paths:
+            if can_write_dir(path):
```

Comment:
Your use case specifically mention full or unhealthy disks as the reason for wanting fallback PEX_ROOT values. This test tests neither of those things. The implication of this PR is that you manually - or via some automation on your nodes - change filesystem permissions on disks that have been determined to disk full or have gone unhealthy. My expecatation was that you'd be changing the test here to actually attempt a mkdir to detect disk full.

If you are in fact doing the permission change on your nodes then you could just be using a PEX rc file on those nodes and changing that file too. This [documentation ](https://pex.readthedocs.io/en/v2.1.112/buildingpex.html?highlight=pexrc#tailoring-pex-execution-at-build-time) is only partly correct but the CLI help, which should always be your primary reading material, is accurate:
```
$ pex --help | grep -A5 pexrc
  --rcfile RC_FILE      An additional path to a pexrc file to read during
                        configuration parsing, in addition to reading
                        `/etc/pexrc` and `~/.pexrc`. If
                        `PEX_IGNORE_RCFILES=true`, then all rc files will be
                        ignored. (default: None)
```

So you could just drop in an `/etc/pexrc ` ith contents:
```
PEX_ROOT=/disk1/.pex
# PEX_ROOT=/disk2/.pex
```
And edit the file when `/disk1` fails.

Reply:
We are not doing permission change. All user can configure is `PEX_ROOT` and currently user application would fail immediately if there is a disk issue. As I explained in https://github.com/pantsbuild/pex/issues/1964, we won't fix the node unless all disk are unhealthy.

Reply:
@manuzhang I'm confused. I have not tested this, but if a disk is full and it has a directory you have write permissions to that was established prior to the disk full condition, I'm pretty sure a permissions check will pass; but later, when you go to use that directory to write files into, those writes will fail - not due to a permissions issue, but due to a disk full. If that's right, that's why I suggest this PR is confusing, since to test for both write permissions and not disk full, you'd have to change the test to an `os.mkdir`. Even after doing that though - since the directory may have been created long before disk full, the `os.mkdir` will no-op successfully and you will still not have tested the disk full condition!

Reply:
@jsirois Permission issue and disk full are two cases. I didn't check the details of `can_write_dir` and thought it returned false whenever the disk couldn't be written to.  Thanks for patiently explain the race condition. Now I understand it and think it would be better to fall back on exceptions. WDYT?

Reply:
I think you still need to spell out some details. How do you propose you trigger those exceptions when calculating which pex root to use from the list of pex roots? Do you try to write a 1 byte file and then delete it if successful and proceed? Do you try to write a file that's bigger than that to attempt to ensure the whole PEX run will succeed - if so, what size do you guess at? Say the guess is 100MB. That's pretty big so writing a file full of 0s - say - and then deleting it will take some time. I wouldn't want to force that time - say its 50ms on all users. Pants for one would certainly not like this at all!

You have a very thorny problem here in my opinion to attempt to solve automatically. If this were my problem, from the information you've given so far, I'd personally write a cron job that pre-emptively checked for disk health / impending disk full - lets say every 10 minutes - and I'd have the job re-write `/etc/pexrc` if it determined a disk was going bad or getting within some threshold of disk full.

Reply:
We don't trigger those exceptions or detect disk issues proactively. I mean we fall back on user exceptions.

Reply:
I understand that you want to do that, I'm suggesting its nearly impossible. Pro-actively managing a machine is not. That you can do with a cron job like I suggested or via other means. This is a very standard thing to do! Many machines have /tmp cleaner cron jobs for example to clear up disk (or memory) space used by old temp files for just this sort of reason.

Reply:
Maybe it makes sense for you to really detail the exact use case fully on the ticket. Whose machines are these? Can you install cron jobs or can you not for some reason? Why do you encounter disk fulls? How often do you encounter bad disks? Etc...

Reply:
I've added more details in the issue comments.